src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,186 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            setHead(value);
+            return;
+        }
+        Node<T> last = tail;
+        //Node<T> newNode =
+        tail = new Node<>(last, value, null);//newNode;
+        if (last == null) {
+            head = tail;
+        } else {
+            last.next = tail;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (!(index >= 0 && index <= size)) {
+            throw new IndexOutOfBoundsException(
+                    "Index passed to the method is negative or "
+                            + "bigger than size of MyLinkedList. "
+                            + "Index: " + index + ", Size: " + size);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            setHead(value);
+            return;
+        }
+        Node<T> currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
+
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        currentNode.prev.next = newNode;
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = head;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
```

Comment:
you have the same logic in several places, create a separate method that will be looking for node by index